Index: mypasswordsafe-20091018/src/pwsafe/Util.h
===================================================================
--- mypasswordsafe-20091018/src/pwsafe/Util.h
+++ mypasswordsafe-20091018/src/pwsafe/Util.h	2010-01-08 07:41:25.950389845 +0200
@@ -171,7 +171,7 @@
 }
 #endif
 
-inline char * strFind( const char *str, const char *fstr )
+inline const char * strFind( const char *str, const char *fstr )
 {
 	return strstr( str, fstr );
 }
